doc/source/serve/tutorials/gradio-integration.md

```diff
@@ -0,0 +1,137 @@
+# Scaling your Gradio App with Ray Serve
+
+In this guide, we will show you how to scale up your Gradio Application using Ray Serve. There is no need to change the internal architecture of your Gradio App; instead, we will neatly wrap it in a Ray Serve deployment which can then be scaled up to multiple replicas with access to more resources.
+
+## Dependencies
+
+To follow this tutorial, you will need Ray Serve and gradio. If you haven't already, install them by running:
+```bash
+$ pip install "ray[serve]"
+$ pip install gradio
+```
+For the purposes of this tutorial, we will be working with Gradio Apps that runs text summarization and text generation models. **Note that you can substitute this Gradio App for any Gradio App of your own!**
+
+We will be using HuggingFace's pipeline to access the model. First, let's install the transformers module.
+```bash
+$ pip install transformers
+```
+
+## Basic Example: Deploy your Gradio App with Ray Serve
+
+In this example, we will show you the easy, straightforward way to deploying your app onto Ray. Start by creating a new python file named `demo.py` and import all dependencies.
+```{literalinclude} ../../../../python/ray/serve/examples/doc/gradio-integration.py
+:start-after: __doc_import_begin__
+:end-before: __doc_import_end__
+```
+
+Then, we construct the (optional) Gradio app `io`:
+> **Note**: Remember you can substitute this with your own Gradio app if you want to try scaling up your own Gradio app!
+```{literalinclude} ../../../../python/ray/serve/examples/doc/gradio-integration.py
+:start-after: __doc_gradio_app_begin__
+:end-before: __doc_gradio_app_end__
+```
+
+
+### Understanding `GradioServer`
+In order to deploy your Gradio App onto a Ray Cluster, you need to wrap your Gradio App in a serve deployment. `GradioIngress` acts as that wrapper. It serves your Gradio App on a Ray cluster.
+```python
+@serve.deployment
+class GradioServer:
+    def __init__(self, io: gr.Blocks):
+        self.app = gr.routes.App.create_app(io)
+
+    async def __call__(self, request: starlette.requests.Request):
+        sender = ASGIHTTPSender()
+        await self.app(request.scope, receive=request.receive, send=sender)
+        return sender.build_asgi_response()
+```
+
+### Deploy your Gradio Server
+You can increase the number of replicas of your application (which can take on more client requests) or increase the number of CPUs and/or GPUs available to each replica.
+
+Then, using either the example we created above, or an existing Gradio App (of type `Interface`, `Block`, `Parallel`, etc.), wrap it in your Gradio Server.
+
+```{literalinclude} ../../../../python/ray/serve/examples/doc/gradio-integration.py
+:start-after: __doc_app_begin__
+:end-before: __doc_app_end__
+```
+
+Finally, deploy your Gradio Server!
+```bash
+$ serve run demo:app
+```
+
+Now you can access your Gradio app at `http://localhost:8000`! This is what it should look like:
```

Comment:
Can we link to ["Deploying Ray Serve"](https://docs.ray.io/en/releases-1.13.0/serve/deployment.html) or ["Putting Ray Serve Deployment Graphs in Production"](https://docs.ray.io/en/master/serve/production.html#id1), so users can learn how to actually deploy this in production?